### PR TITLE
Add ^x,y,z relative offset support to the offset parser

### PR DESCRIPTION
**Pull request by me4502**
Adds a `^` prefix to the offset parser that signifies the vector is a `forwards,up,left` offset rather than raw x/y/z
 
Fixes https://github.com/EngineHub/WorldEdit/issues/1442

### CODE REVIEW

File: worldedit-core/src/main/java/com/sk89q/worldedit/command/argument/OffsetConverter.java [hunk 1]
```diff
@@ -62,6 +67,11 @@ public Component describeAcceptableArguments() {
 
     @Override
     public List<String> getSuggestions(String input, InjectedValueAccess context) {
+        if (input.startsWith("^") && context.injectedValue(Key.of(Player.class)).isPresent()) {
```
Comment (wizjany) [inline worldedit-core/src/main/java/com/sk89q/worldedit/command/argument/OffsetConverter.java:70]: any reason we require `Player` here? would any `Locatable` sender do? (even tho commandblocks' pitch/yaw are probably always `0,0` in theory mobs can be commandsenders too though we don't support them yet)

File: worldedit-core/src/main/java/com/sk89q/worldedit/command/argument/OffsetConverter.java [hunk 2]
```diff
@@ -70,7 +83,47 @@ public Component describeAcceptableArguments() {
...
+                m1.getZ(), m2.getZ(), m3.getZ(), 0
+            );
+
+            return vectorConverter.convert(input.substring(1), context).map(blockVector3s ->
+                blockVector3s.stream().map(vector -> transform
+                    .apply(vector.toVector3())
+                    // This *MUST* be rounded before converting to block points
+                    .round().toBlockPoint()
+                ).collect(Collectors.toList())
+            );
```
Comment (octylFractal) [inline worldedit-core/src/main/java/com/sk89q/worldedit/command/argument/OffsetConverter.java:123]: Extract to a method.

File: worldedit-core/src/main/java/com/sk89q/worldedit/math/MathUtils.java [hunk 3]
```diff
@@ -66,11 +66,10 @@ public static double dCos(double degrees) {
                 case 0:
                     return 1.0;
                 case 90:
+                case 270:
```
Comment (octylFractal) [inline worldedit-core/src/main/java/com/sk89q/worldedit/math/MathUtils.java:69]: Revert these changes.

**Review by octylFractal — CHANGES_REQUESTED** [review]

File: worldedit-core/src/main/java/com/sk89q/worldedit/command/argument/OffsetConverter.java [hunk 4]
```diff
@@ -62,15 +69,66 @@ public Component describeAcceptableArguments() {
...
+    }
+
     @Override
     public ConversionResult<BlockVector3> convert(String input, InjectedValueAccess context) {
-        return directionVectorConverter.convert(input, context)
-            .orElse(vectorConverter.convert(input, context));
+        if (input.startsWith("^")) {
+            // Looking at a relative vector.
+            Actor actor = context.injectedValue(Key.of(Actor.class))
+                .orElseThrow(() -> new IllegalStateException("An actor is required to use relative offsets"));
```
Comment (octylFractal) [inline worldedit-core/src/main/java/com/sk89q/worldedit/command/argument/OffsetConverter.java:116]: oh yea, you're uh, not supposed to throw here. Return a `FailedConversion.from(ex)` instead, or just wrap the whole thing in a `try-catch` :P

**Review by octylFractal — CHANGES_REQUESTED** [review]

**Review by octylFractal — APPROVED** [review]

**Pull request merged**